### ISSUE TITLE
Refactor Results Directory Structure for Lunar Lander Model

### ISSUE THREAD

**Issue by bohoro** [bohoro/ZeroPlayM4#1]
### Problem Description

Currently, the training checkpoints, logs, and evaluation results for the Lunar Lander reinforcement learning model are not consistently saved into organized subfolders within the `results/` directory. This leads to:

* **Difficulty in tracking experiments:** It's hard to quickly identify which files belong to a specific training run, hyperparameter configuration, or model version.
* **Cluttered `results/` directory:** The top-level `results/` folder becomes a dumping ground, making it challenging to navigate and find relevant information.
* **Increased risk of overwriting:** Without clear separation, there's a higher chance of accidentally overwriting previous experiment data.
* **Poor reproducibility:** Replicating past results becomes more challenging when the associated artifacts are scattered.

### Proposed Solution

We need to implement a structured directory system within `results/` to ensure all output files are logically organized. The proposed structure is as follows:

```
results/
├── lunar_lander/
│   ├── [experiment_name_or_timestamp]/  <-- e.g., '20250524_ppo_run1', 'hyperparam_tune_gamma_0.99'
│   │   ├── checkpoints/
│   │   │   ├── model_epoch_X.pt
│   │   │   └── best_model.pt
│   │   ├── logs/
│   │   │   ├── tensorboard/
│   │   │   │   ├── events.out.tfevents...
│   │   │   ├── training.log
│   │   │   └── eval.log
│   │   ├── plots/
│   │   │   ├── rewards_over_time.png
│   │   │   └── episode_lengths.png
│   │   ├── config.yaml          <-- Snapshot of hyperparameters for this run
│   │   └── metrics.json         <-- Final evaluation metrics
│   ├── [another_experiment_name]/
│   │   ├── ...
```

**Key changes to be implemented:**

1.  **Dynamic Subfolder Creation:**
    * Implement a mechanism (e.g., using `datetime` or a unique experiment ID) to create a new subfolder for each training run within `results/lunar_lander/`. This subfolder name should ideally be descriptive (e.g., `YYYYMMDD_HHMMSS_algorithm_variant`).
2.  **Dedicated Checkpoints Folder:**
    * All model checkpoints (`.pt`, `.pth`, etc.) should be saved into `results/lunar_lander/[experiment_name]/checkpoints/`.
3.  **Dedicated Logs Folder:**
    * All training logs (TensorBoard event files, text logs) should go into `results/lunar_lander/[experiment_name]/logs/`.
    * Consider a further `tensorboard/` subfolder within `logs/` for clarity if TensorBoard is used.
4.  **Dedicated Plots Folder (Optional but Recommended):**
    * If any plots are generated during or after training (e.g., reward curves), they should be saved to `results/lunar_lander/[experiment_name]/plots/`.
5.  **Configuration Snapshot:**
    * Save a copy of the exact configuration/hyperparameters used for each run (e.g., `config.yaml` or `config.json`) directly into `results/lunar_lander/[experiment_name]/`. This is crucial for reproducibility.
6.  **Final Metrics Summary:**
    * Save a summary of final evaluation metrics (e.g., average reward, success rate) to a file like `metrics.json` within `results/lunar_lander/[experiment_name]/`.

### Acceptance Criteria

* Running `train.py` (or equivalent training script) successfully creates a new, uniquely named experiment subfolder under `results/lunar_lander/`.
* Model checkpoints are saved exclusively within the `checkpoints/` subfolder of the respective experiment directory.
* Training logs (including TensorBoard logs) are saved exclusively within the `logs/` subfolder of the respective experiment directory.
* A configuration file (`config.yaml` or similar) containing the hyperparameters of the run is saved within the experiment directory.
* (Optional) Generated plots are saved within the `plots/` subfolder.
* (Optional) A summary of final evaluation metrics is saved within the experiment directory.

### Technical Notes / Implementation Hints

* **Python `pathlib`:** Consider using `pathlib` for robust path manipulation.
* **Logging Libraries:** If not already using one, consider `logging` module for structured log outputs.
* **Configuration Management:** Libraries like Hydra or simple `argparse` with YAML parsing can help manage and save configurations.
* **TensorBoard `logdir`:** Ensure the `logdir` argument for TensorBoard `SummaryWriter` is correctly set to point to the new `logs/tensorboard/` path.

### Benefits

* Improved organization and navigability of results.
* Enhanced reproducibility of experiments.
* Easier comparison between different training runs.
* Reduced risk of data loss or overwriting.

---

**Why these changes improve the issue:**

* **More Descriptive Title:** "Refactor Results Directory Structure for Lunar Lander Model" clearly states the intent.
* **Labels:** Helps categorize and filter issues. `good first issue` can attract new contributors.
* **Problem Description:** Elaborates on *why* this is a problem, highlighting the negative impacts.
* **Proposed Solution:**
    * Provides a clear, visual directory structure.
    * Breaks down the solution into specific, actionable "Key changes to be implemented" points.
    * Suggests concrete file names and types (e.g., `config.yaml`, `metrics.json`).
* **Acceptance Criteria:** Defines what "done" looks like in a measurable way. This is crucial for testing and verifying the fix.
* **Technical Notes / Implementation Hints:** Offers guidance to the person who will be implementing the fix, making it easier for them to get started.
* **Benefits:** Re-iterates the positive outcomes, reinforcing the value of addressing the issue.

This revised issue is much more comprehensive and helpful for anyone trying to understand and resolve the problem.
